cmd/forwarder/fixer.go

```diff
@@ -61,6 +61,15 @@ func getMetadata(req *http.Request, cred *credential, metadataId string) ([]byte
 	return metadataWriter.Bytes(), foundMetadata
 }
 
+// Truncation function to fix header.Hostname <=48 characters
+// Returns:
+// * string truncated to 48 characters
+// * integer representing the amount of truncated characters - not yet implemented
+func truncate(hostname []byte) ([]byte) {
+	m := hostname[0:48]
+	 return m
+}
```

Comment:
You need a better method than this, with a better name. This method as-is has a few problems:

1. It's called "truncate", but it's really only useful for the Hostname field, because it specifically truncates *to 48 characters*.
2. It causes you have to repeat "48" here and also in the code that *invokes* the method, which is non-obvious to readers of the code.

I'll give you a method signature to use, along with comments describing it, but I won't implement the body of it. Use this:

```go
 // Write a header field into the messageWriter buffer. Truncates to maxLength. 
  // Returns true if the input string was truncated, and false otherwise.
 func writeField(str []byte, int maxLen, bytes.Buffer* messageWriter) (bool) {
 }
```

You can ignore the return value when you invoke this function. Later when we're adding metrics, we'll use it. But for now, throw it away like this:

```go
   _ = writeField(header.Hostname, 48, &messageWriter)
```